crates/fluvio-cli/src/client/smartmodule/generate.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use fluvio_extension_common::Terminal;
+use clap::Parser;
+use cargo_generate::{GenerateArgs, TemplatePath, generate};
+use fluvio::Fluvio;
+
+use crate::Result;
+use crate::client::cmd::ClientCmd;
+use crate::error::CliError;
+
+const GIT_TEMPLATE: &str = "https://github.com/infinyon/fluvio-smartmodule-template";
+
+/// Create a new SmartModule with a given name
+#[derive(Debug, Parser)]
+pub struct GenerateSmartModuleOpt {
+    /// The name of the SmartModule to create
+    name: String,
+}
+
+#[async_trait]
+impl ClientCmd for GenerateSmartModuleOpt {
+    async fn process_client<O: Terminal + Debug + Send + Sync>(
+        self,
+        _out: Arc<O>,
+        _fluvio: &Fluvio,
+    ) -> Result<()> {
+        let template_path = TemplatePath {
```

Comment:
Neither this `struct` nor `GenerateArgs` implements Default trait. So I had to provide initial values for both.
I'll open a PR to the generate crate to see if they allow us to have a `Default` implementation.